### DISCUSSION TITLE
[bug]: Unable to transfer using sendTransaction in IOS

### DISCUSSION THREAD

**Discussion by souravsoftwareer** [WalletConnect/walletconnect-monorepo#3490] [Developer Support]
### Description

I am getting issue in transfer. it is working fine in Android no issue at all.But in IOS transfer is not happening it gives following error
![1691561989231 (1)](https://github.com/WalletConnect/react-native-examples/assets/8158259/55e75e5b-b8a4-4d9a-8e89-76ad1931ebee)

https://github.com/WalletConnect/react-native-examples/assets/8158259/0b4778ec-7a89-4cb9-bd17-7a05713e8db0

 

### Issue Environment

- [X] dapps/v2Explorer
- [ ] wallets/rn_cli_wallet
- [ ] wallets/rn_cli_wallet_068_5
- [ ] wallets/expo_wallet
- [ ] wallets/web3wallet_tutorial
- [X] In my own code

### Output of `npx react-native info`

info Fetching system and libraries information...
System:
  OS: macOS 13.3
  CPU: (12) x64 Intel(R) Core(TM) i5-10400 CPU @ 2.90GHz
  Memory: 1.51 GB / 32.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 17.0.0
    path: ~/.nvm/versions/node/v17.0.0/bin/node
  Yarn:
    version: 1.22.19
    path: /usr/local/bin/yarn
  npm:
    version: 8.1.0
    path: ~/.nvm/versions/node/v17.0.0/bin/npm
  Watchman: Not Found
Managers:
  CocoaPods:
    version: 1.12.1
    path: /Users/john/.rvm/rubies/ruby-2.7.6/bin/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 22.4
      - iOS 16.4
      - macOS 13.3
      - tvOS 16.4
      - watchOS 9.4
  Android SDK: Not Found
IDEs:
  Android Studio: 2022.2 AI-222.4459.24.2221.9971841
  Xcode:
    version: 14.3/14E222b
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.7
    path: /usr/bin/javac
  Ruby:
    version: 2.7.6
    path: /Users/john/.rvm/rubies/ruby-2.7.6/bin/ruby
npmPackages:
  "@react-native-community/cli": Not Found
  react:
    installed: 18.2.0
    wanted: 18.2.0
  react-native:
    installed: 0.72.1
    wanted: 0.72.1
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: false
iOS:
  hermesEnabled: true
  newArchEnabled: false

### Steps to reproduce

Make connection
Then try to send eth in trust wallet
Got following issue
https://github.com/WalletConnect/react-native-examples/tree/main/dapps/v2Explorer

### Snack, code example, screenshot, or link to a repository

https://github.com/WalletConnect/react-native-examples/tree/main/dapps/v2Explorer

**Comment by souravsoftwareer** [comment]
Device configuration
![Skype_Picture_2023_08_09T06_27_47_436Z](https://github.com/WalletConnect/react-native-examples/assets/8158259/b6418960-6652-47c0-abc4-c25cee56ed87)

**Comment by souravsoftwareer** [comment]
how to join discord web3modal channel

**Comment by ignaciosantise** [comment]
hey @souravsoftwareer 👋 Apparently Trust Wallet is not working as expected. Can you tell me which version of Trust are you using on Android and iOS?

**Comment by souravsoftwareer** [comment]
It is working well in android but i am having issue in IOS trust wallet i am using trust wallet version as follows
9.2 it is latest version of trust wallet. @ignaciosantise kindly look at this.

Trust wallet app version using
Android version : 7.29.2
IOS version : 9.2

**Comment by souravsoftwareer** [comment]
I am transferring ETH token i have balance as well you can check screenshot below.
![1691644908085](https://github.com/WalletConnect/react-native-examples/assets/8158259/0a55cfc3-4c01-4389-a441-29427b86bc0e)

**Comment by ignaciosantise** [comment]
@souravsoftwareer Apparently Trust added some validations to the request. I added new parameters to the transaction to solve this -> you can check them here https://github.com/WalletConnect/react-native-examples/pull/93

**Comment by souravsoftwareer** [comment]
@ignaciosantise do i need to download example again ?

**Comment by ignaciosantise** [comment]
no, just check the code and update it locally

**Comment by souravsoftwareer** [comment]
@ignaciosantise have you changed in this file
https://github.com/WalletConnect/react-native-examples/pull/93/commits/e586f7f9e1cf4ec6ae547a0d5d7bc196dc7054d7 ?

**Comment by ignaciosantise** [comment]
yes

**Comment by souravsoftwareer** [comment]
okay i will check and let you know.

**Comment by souravsoftwareer** [comment]
@ignaciosantise this issue is resolved but i have another issue i have tried to send MANA token but it is showing 0th over trust wallet i have send 9 from amount i am using following code for mana token 

 console.log("erc 20 coin")

        if (!web3Provider) {
          throw new Error('web3Provider not connected');
        }
        provider?.enable()

        const { chainId } = await web3Provider.getNetwork();
        if(chainId!=1) {
          await updateSessionMethod(chainId)
        }

        let web3 = new Web3(provider)
        let tokenAddress = item?.contract_address
        let minABI = transferErcAbi
        let decimals = item?.decimals ?? 18;

        let contract = new web3.eth.Contract(minABI, tokenAddress);
        console.log("decimals ==>", decimals)
        // let value = Web3Utils.toWei(amount, "ether")
        let value = sanitizeHex(numberToHex(parseFloat(amount)));
        
        let  value1 = ethers.utils.parseUnits(amount, decimals);

        contract.methods.transfer(walletAddress, Platform.OS == 'android'?value1:value)
          .send({ from: address })
          .on('transactionHash', async (resultt) => {

            // record a transaction
            console.log("result", resultt);
            sendEmailTransactionData(resultt, amount, item.tokenSymbol)
            await setAsyncItem(strings.OUT_FROM_HOME, "1")
            props.navigation.navigate('TransactionDetail', {
              txId: resultt,
              amount,
              destination: walletAddress,
              source: address,
              item: item
            })
          })
          .on('receipt', (receiptt) => {
            console.log("receipt", receiptt)
            // update transaction by hash
            let editObj = {
              trx_id: receiptt.transactionHash,
              status: 1,
              response_data: JSON.stringify(receiptt)

            }
          })
          .on('error', (err: any) => {

            setLoader(false)
            toast.error({ message: "Transaction Declined !!! Please try after some time" })
          })
          
          I have made video for it as well please check following link i have linked file as well in which transfer is working.
          You can check transferCoin() function i have used for.
          
          https://drive.google.com/file/d/1NUzW-cloVJEjftZjIDZDzgYzleG8wKvK/view?usp=sharing
          
          File link
          
          https://drive.google.com/file/d/1_ogJQU7mPNqdd6SKbBGhYuCfZpxwOieP/view?usp=sharing

**Comment by souravsoftwareer** [comment]
hi @ignaciosantise kendly reply back

**Comment by souravsoftwareer** [comment]
Hi @ignaciosantise kindly reply back to me  it is very important

**Comment by souravsoftwareer** [comment]
@ignaciosantise also check meta mask not connecting

**Comment by glitch-txs** [comment]
> @ignaciosantise this issue is resolved but i have another issue i have tried to send MANA token but it is showing 0th over trust wallet i have send 9 from amount i am using following code for mana token

Hi, is your code public? could you share a minimal example of it?

**Reply by souravsoftwareer** [reply]
Hi kindly check  video of example
     https://drive.google.com/file/d/1NUzW-cloVJEjftZjIDZDzgYzleG8wKvK/view?usp=sharing
      
      And this is file link where i am using the code for transfer file name is Transfer
      
      https://drive.google.com/file/d/1_ogJQU7mPNqdd6SKbBGhYuCfZpxwOieP/view?usp=sharing

**Reply by glitch-txs** [reply]
could you provide a [MRE](https://en.wikipedia.org/wiki/Minimal_reproducible_example#:~:text=In%20computing%2C%20a%20minimal%20reproducible,to%20be%20demonstrated%20and%20reproduced.)?

**Reply by souravsoftwareer** [reply]
can you provide your apple id i can add you over testflight ?

**Comment by souravsoftwareer** [comment]
Please check this demo
https://drive.google.com/file/d/1xhyZAQ0o1s7IF6HjoLFVzXt3AiSX-DXz/view?usp=sharing

I have added sendTransaction1 function over there mana token setup done please check this needfull

**Comment by souravsoftwareer** [comment]
Make connection and then try to use sendTransaction

**Comment by souravsoftwareer** [comment]
Check with trust wallet

**Comment by souravsoftwareer** [comment]
Hi Wallet Connect
Kindly check android video how it is showing mana and dodo token when transferring token it is showing mana token amount how many i am transferring it is working fine in android kindly check video link below same thing i want to do in IOS as well.
Please guide me to do in IOS 
https://drive.google.com/file/d/1g50zjtAPOy0uBm7skBxw8RwSwBwajIgL/view?usp=sharing
Android erc20 address with trust wallet
0xa05bF59B49FF4BC3eddeA7343a44a7a5340bA853

Please have a look on IOS video it is showing us -0eth while i transfer MANA or DODO token. This should show MANA or DODO token which ever i am going to transfer from Rhombus wallet on Trust Wallet.In android above video it works fine.
https://drive.google.com/file/d/1681hSsDuRvuAb_tRZ4v9vve_sDj5eMN_/view?usp=sharing

IOS erc20 address with trust wallet
0xa05bF59B49FF4BC3eddeA7343a44a7a5340bA853

**Comment by mshahjahan-mufin** [comment]
@glitch-txs @ignaciosantise 

Can you update Flutter SDK as well I am facing the same issue.

**Reply by souravsoftwareer** [reply]
@ShahjahanMirza92 i am using wallet connect v2 sdk do i need to update that ?

**Reply by souravsoftwareer** [reply]
I am using react-native

**Reply by mshahjahan-mufin** [reply]
I think so, as in the example by @ignaciosantise he update something. Also, let me know the issues you are facing. I might have faced similar problems and might be able to guide you in the correct direction.

**Reply by souravsoftwareer** [reply]
@shahji-mufin it is not working after updating wallet connect v2
